spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/abstractnn/AbstractModule.scala

```diff
@@ -73,6 +73,44 @@ abstract class AbstractModule[A <: Activity: ClassTag, B <: Activity: ClassTag,
    */
   var gradInput: A = Activity.allocate[A, T]()
 
+
+  private[bigdl] var inputShapeValue: Activity = null
+
+  private[bigdl] var outputShapeValue: Activity = null
+
+  /**
+   * There's no batch dim in the inputShape which just represent a sample record.
+   */
+  def getInputShape(): Activity = {
+    inputShapeValue
+  }
+
+  /**
+   * There's no batch dim in the outputShape which just represent a sample record.
+   */
+  def getOutputShape(): Activity = {
+    if (outputShapeValue == null) {
+      throw new RuntimeException("You should build this model first before getting the outputshape")
+    }
+    outputShapeValue
+  }
+
+  def isBuilt(): Boolean = {
+    getOutputShape() != null
+  }
+
+  private[bigdl] def build(inputShape: Activity): Unit = {
+    this.outputShapeValue = computeOutputShape(inputShape)
+    this.inputShapeValue = inputShape
+  }
+
+  /**
+   * There's no batch dim in the inputShape which just represent a sample record.
+   */
+  def computeOutputShape(inputShape: Activity): Activity = {
```

Comment:
this should be a private method, right?

Reply:
not sure if we want to expose such useful feature to user. I'm OK to private this method.

Reply:
I don't think this is useful for user. If user want to see output shape, they should call getOutputShape, right?